core/main/src/processor/main_context_processor.rs

```diff
@@ -176,16 +175,8 @@ impl MainContextProcessor {
 
             if state.supports_app_catalog() {
                 let ignore_list = vec![state.get_device_manifest().applications.defaults.main];
-
-                let client = state.get_client().clone();
+                let client = state.get_client();
                 client.add_event_processor(AppsUpdater::new(client.get_extn_client(), ignore_list));
-
```

Comment:
Moved to the extension so that it is processed after the extension has initialized and is able to. Should be done by the extension anyway since it sets up an extension listener thread internally.